pytorch_lightning/trainer/trainer.py

```diff
@@ -1537,14 +1537,20 @@ def _call_accelerator_hook(
         *args: Any,
         **kwargs: Any,
     ) -> Optional[Any]:
-        self.lightning_module._current_fx_name = hook_name
+        pl_module = self.lightning_module
```

Comment:
why do we have to do all this, since previously we just had one line? 
https://github.com/PyTorchLightning/pytorch-lightning/blob/3d6262b7a91215e72019d720e742e6261e1636dc/pytorch_lightning/loops/epoch/evaluation_epoch_loop.py#L220

https://github.com/PyTorchLightning/pytorch-lightning/blob/3d6262b7a91215e72019d720e742e6261e1636dc/pytorch_lightning/loops/optimization/manual_loop.py#L105

Reply:
Previously was incomplete, we should always set and unset

Reply:
This is required for some nested hooks such as training_step with manual optimization.